### DISCUSSION TITLE
How to refer to query string parameters in custom result template/layout

### DISCUSSION THREAD

**Discussion by koenti** [microsoft-search/pnp-modern-search#2152] [Ideas]
**example:** SharePoint page with URL https://<mysite>/sitepages/mypage.aspx?box=ABC.  On that page a PnP Search result webpart with custom result template/layout.

How can I refer to a query string parameter in my result template?  I want to use the box value (ABC) in my result template.

Tokens such as {QueryString.box} are not exposed?

**Comment by wobba** [comment]
Use `@root.inputQueryText`. If you switch to the debug template you see the full object model of what you can reference.

**Reply by koenti** [reply]
Hi,
I need the URL query string parameter.  You refer to inputQueryText.
In debug template I don't see any **URL query string parameter** value.
Can we reuse the "[context-tokens](https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/#context-tokens)" in a template?  eg {{QueryString.box}}

**Reply by wobba** [reply]
I'll tag this as a feature request, and register a helper to give access to the window.location object. Something like https://github.com/Automattic/wp-calypso/blob/a088d37d92ff3cb80c5a4dbc70183408a4249ffe/packages/calypso-analytics/src/tracks.ts#L65

**Reply by koenti** [reply]
Having access to **window.location** is great.  
With {{urlParser}}-helper we can extract everything:
{
	"protocol": "https:",
	"slashes": true,
	"auth": null,
	"host": "xxx.sharepoint.com",
	"port": null,
	"hostname": "xxx.sharepoint.com",
	"hash": null,
	"search": "?box=ABC",
	"query": "box=ABC",
	"pathname": "/sites/test",
	"path": "/sites/test/sitepages/page.aspx?box=ABC",
	"href": "https://xxx.sharepoint.com/sites/test/sitepages/page.aspx?box=ABC"
}
Can you estimate when this can be made available?

**Reply by wobba** [reply]
I have an open PR for this #2167. See the updated docs for usage. I made support for `urlParser` to use current URL and added a new helper `getUrlParameter`.
